pandas/io/parsers.py

```diff
@@ -2223,11 +2223,16 @@ def _clean_index_names(columns, index_col):
 def _get_empty_meta(columns, index_col, index_names):
     columns = list(columns)
 
-    if index_col is not None:
+    # bool type inherits from int
+    if isinstance(index_col, int) and index_col is not False:
+        index_col = [ index_col ]
+
+    if index_col:
+        index_col = list(index_col)
```

Comment:
Wouldn't that fail for `index_col=1` or`index_col=[]`?

Also - I added `index_col = list(index_col)` in case an iterator is passed in, even though the documentation doesn't specify that this is supported.  Maybe that's not necessary?